### ISSUE TITLE
`ignores` not being respected with unused disable directive reporting

### ISSUE THREAD

**Issue by brettz9** [eslint/eslint#17930]
### Environment

Node version: 18.16.0
npm version: 9.5.1
Local ESLint version: 9.0.0-alpha.0
Global ESLint version:  8.55.0
Operating System: Mac 14.2.1

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
export default [
  {
    ignores: ['dist/**/*.js', '.ignore/**/*.js'],
  }
];
```
</details>

Within `dist/`:

```js
/* eslint-disable complexity */
```

### What did you expect to happen?

No errors to be reported

### What actually happened?

Errors were reported for unused disable directives

### Link to Minimal Reproducible Example

https://

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

I cannot reproduce in the playground, but this should be easy to reproduce.

**Comment by fasttime** [comment]
Hi @brettz9, thanks for the report. I can't seem to reproduce the problem, see this repro: https://stackblitz.com/edit/stackblitz-starters-sj8b1u. Can you check what I'm missing?

**Comment by snitin315** [comment]
@fasttime I'm able to reproduce it with ESLint v8, in the repro you shared if we change eslint to v8 and run `npm run lint -- --report-unused-disable-directives`

<img width="747" alt="Screenshot 2023-12-31 at 9 34 01 AM" src="https://github.com/eslint/eslint/assets/46647141/4017b339-7fd0-423f-b6b2-31508e46b8fb">





However, this is not reproducible with ESLint v9

<img width="783" alt="Screenshot 2023-12-31 at 9 30 14 AM" src="https://github.com/eslint/eslint/assets/46647141/723a1086-fc59-4e9e-a8a1-3ad84f8f8af9">

**Comment by fasttime** [comment]
Thanks @snitin315, that makes sense. I've switched the repro to use ESLint 8.55.0 and added `--report-unused-disable-directives` to the npm `lint` script, but I'm still not getting an error.

![image](https://github.com/eslint/eslint/assets/6367844/59e6b3ba-1682-44c0-8895-ce38794ad280)

I'm also not getting an error when running this locally with ESLint 9.0.0-alpha.0 having installed ESLint 8.55.0 globally. Maybe an issue with outdated dependencies?

**Comment by brettz9** [comment]
My sincere apologies. I hadn't reduced the test case down far enough.

It appears the issue is that I was importing a plugin that was also in `dist` (I wanted to apply the linter plugin in my repo to the repo itself). See https://stackblitz.com/edit/stackblitz-starters-x6plq4?file=dist%2Findex.js

**Comment by fasttime** [comment]
Thanks for the update @brettz9. So your config looks like this:

```js
import index from './dist/index.js';

export default [
  {
    plugins: {
      index,
    },
    ignores: ['dist/**/*.js', '.ignore/**/*.js'],
  },
];
```

And ESLint is printing a warning after the upgrade to v9.0.0-alpha.0. This is happening because of [this change](https://github.com/eslint/eslint/pull/17879) in v9 that enables reporting unused disable directives by default. Since the `complexity` rule is not enabled, ESLint will now report a warning when it lints the file with the `eslint-disable` directive.

If the intent was to exclude that file from linting, the `ignores` setting should be moved into a new config element without any other properties:

```js
import index from './dist/index.js';

export default [
  {
    ignores: ['dist/**/*.js', '.ignore/**/*.js'],
  },
  {
    plugins: {
      index,
    },
  },
];
```

This is explained in more detail in [the documentation](https://eslint.org/docs/next/use/configure/ignore).

To keep the current config and just exclude the `index` plugin when linting files located in `dist/` or `.ignore/`, run ESLint with the flag `--report-unused-disable-directives-severity=off`, or set the linter option `reportUnusedDisableDirectives` to `"off"` in your config (see https://eslint.org/docs/next/use/configure/configuration-files#configuration-objects).

**Comment by brettz9** [comment]
Great, thank you very much.
